src/terrain/terrain.cpp

```diff
@@ -915,20 +914,21 @@ Terrain::get_path(
 uint32_t
 Terrain::get_voxel(int x, int y, int z) const {
     // using static ints to prevent dereferencing
-    // The previous material id, and color id are cashed so that materials do 
+    // The previous material id, and color id are cashed so that materials do
     // not need to be dereferenced, and searched through.
-    static unsigned int get_voxel_mat_id = 0;
-    static unsigned int get_voxel_color_id = 0;
-    static uint32_t get_voxel_out_color = 0;
+    static uint8_t get_voxel_previous_mat_id = 0;
+    static uint8_t get_voxel_previous_color_id = 0;
+    static uint32_t get_voxel_previous_out_color = 0;
```

Comment:
Why are all of these prefixed with `get_voxel`? You know that GCC will namespace them for you by function

Also use a variable for `AIR`
```suggestion
    static uint8_t previous_mat_id = AIR;
    static uint8_t previous_color_id = AIR;
    static uint32_t previous_out_color = AIR;
```